### ISSUE TITLE
Chat annotations ("gif embeds") are blank 

### ISSUE THREAD

**Issue by akhepcat** [EionRobb/purple-googlechat#68]
Even as of commit b6b824a4764b51316f7be325492575684647e021, gif annotations (are there other types?)  / 
gif-embeds within a google chat sequence are showing as a blank line in the chat.

Debug log shows that the image URL within the annotations block is there:
`"annotations" : [
{
"type" : "URL",
"start_index" : 0,
"length" : 0,
"drive_metadata" : null,
"user_mention_metadata" : null,
"url_metadata" : {
"title" : null,
"snippet" : null,
"image_url" : "https://lh6.googleusercontent.com/proxy/346Jq0h8qPxAVbA_qAO2-ndrpuNHCD6DRGj0cyUtYYhA43JHig5CJHvVBoNBgKbt21z-JzarIp500sWuNWPtKEdIiDQDnVOUqa2MzVBkVYy9dlmK",
"image_height" : "498",
"image_width" : "498",
"url" : {
"url" : "https://media.tenor.com/SvI_M9diCWcAAAAC/so-do-it-markiplier.gif"
},
"domain" : "media.tenor.com",
"should_not_render" : false,
"int_image_height" : 498,
"int_image_width" : 498,
"mime_type" : "image/*",
"gws_url" : {
"url" : "https://www.google.com/url?q=https://media.tenor.com/SvI_M9diCWcAAAAC/so-do-it-markiplier.gif&source=chat&ust=1669152447408000&usg=AOvVaw0RMcyypxdGTRuSRoZVgZV9"
},
"redirect_url" : {
"url" : "https://chat.google.com/api/get_redirect_url?url_redirect_token=XXXXXXX"
},
"gws_url_expiration_timestamp" : 1669152447409000
},
`

it would be nice to at least see that URL, as a link,  instead of a blank line, that way I can at least decided to click on it.

**Comment by BoFiS** [comment]
Yes, this remains an issue, though weirdly if a GIF is sent as an image attachment instead of using the "GIF" button, it goes through the way images do (mostly)

**Comment by EionRobb** [comment]
I think this is fixed now. 

Before, the gif messages were coming through as a zero-length string with a link so I added code to handle for that. 

But then it turns out the messages get updated and re-sent by the server as embedded images as an updated message event, so have added support for edits/updates which doubly helps, with the added bonus of supporting message updates
